packages/market-service/src/coingecko/coingecko.test.ts

```diff
@@ -1,13 +1,129 @@
+    }
+
+    const eth: CoinGeckoMarketCap = {
+      id: 'ethereum',
+      symbol: 'eth',
+      name: 'Ethereum',
+      image: 'https://assets.coingecko.com/coins/images/279/large/ethereum.png?1595348880',
+      current_price: 3459.72,
+      market_cap: 407989270877,
+      market_cap_rank: 2,
+      fully_diluted_valuation: null,
+      total_volume: 17486135198,
+      high_24h: 3605.6,
+      low_24h: 3459.1,
+      price_change_24h: -134.750135067397,
+      price_change_percentage_24h: -3.74881,
+      market_cap_change_24h: -15764031085.668518,
+      market_cap_change_percentage_24h: -3.7201,
+      circulating_supply: 117874980.3115,
+      total_supply: null,
+      max_supply: null,
+      ath: 4356.99,
+      ath_change_percentage: -20.19316,
+      ath_date: '2021-05-12T14:41:48.623Z',
+      atl: 0.432979,
+      atl_change_percentage: 802982.25606,
+      atl_date: '2015-10-20T00:00:00.000Z',
+      roi: {
+        times: 83.32608527170541,
+        currency: 'btc',
+        percentage: 8332.60852717054
+      },
+      last_updated: '2021-10-10T22:16:22.950Z'
+    }
+
+    it('can flatten multiple responses', async () => {
+      mockedAxios.get.mockResolvedValueOnce({ data: [eth] }).mockResolvedValue({ data: [btc] })
+      const result = await getByMarketCap()
+      expect(Object.keys(result).length).toEqual(2)
+    })
+
+    it('can sort by market cap', async () => {
+      mockedAxios.get.mockResolvedValueOnce({ data: [btc] }).mockResolvedValue({ data: [eth] })
+      const result = await getByMarketCap()
+      expect(Object.keys(result)[0]).toEqual(adapters.coingeckoToCAIP19(btc.id))
+    })
+
+    it('can handle rate limiting', async () => {
+      mockedAxios.get.mockResolvedValue({ status: 429 })
+      const result = await getByMarketCap()
+      expect(Object.keys(result).length).toEqual(0)
+    })
+
+    it('can return some results if partially rate limited', async () => {
+      mockedAxios.get.mockResolvedValueOnce({ status: 429 }).mockResolvedValue({ data: [eth] })
+      const result = await getByMarketCap()
+      expect(Object.keys(result).length).toEqual(1)
+    })
+
+    it('can use default args', async () => {
+      mockedAxios.get.mockResolvedValue({ data: [btc] })
+      await getByMarketCap()
+      expect(mockedAxios.get).toHaveBeenCalledTimes(10)
+    })
+
+    it('can use override args', async () => {
+      mockedAxios.get.mockResolvedValue({ data: [btc] })
+      const pages = 1
+      const perPage = 10
+      await getByMarketCap({ pages, perPage })
+      expect(mockedAxios.get).toHaveBeenCalledTimes(1)
+      const url =
+        'https://api.coingecko.com/api/v3/coins/markets?vs_currency=usd&order=market_cap_desc&per_page=10&page=1&sparkline=false'
+      expect(mockedAxios.get).toBeCalledWith(url)
+    })
+
+    it('can map coingecko to caip ids', async () => {
+      mockedAxios.get.mockResolvedValueOnce({ data: [btc] }).mockResolvedValue({ data: [eth] })
```

Comment:
I wouldn't mind some "mockRejectedValue" tests to validate what is expected to happen in the API is down.